deps/rabbit/src/mc.erl

```diff
@@ -223,9 +226,29 @@ routing_headers(#?MODULE{protocol = Proto,
 routing_headers(BasicMsg, Opts) ->
     mc_compat:routing_headers(BasicMsg, Opts).
 
+-spec exchange(state()) -> rabbit_misc:resource_name().
+exchange(#?MODULE{annotations = #{?EXCHANGE := Exchange}}) ->
+    Exchange;
+exchange(BasicMessage) ->
+    mc_compat:get_annotation(?EXCHANGE, BasicMessage).
+
+-spec routing_keys(state()) -> [rabbit_types:routing_key(),...].
+routing_keys(#?MODULE{annotations = #{?ROUTING_KEYS := RoutingKeys}}) ->
+    RoutingKeys;
+routing_keys(BasicMessage) ->
+    mc_compat:get_annotation(?ROUTING_KEYS, BasicMessage).
+
+-spec exchange_and_routing_keys(state()) ->
+    {rabbit_misc:resource_name(), [rabbit_types:routing_key(),...]}.
+exchange_and_routing_keys(#?MODULE{annotations = #{?EXCHANGE := Exchange,
```

Comment:
In the interest of keeping APIs from getting too big it feels a little bit like we could let this function go and just use `mc:exchange/1` and `mc:routing_keys/1`. WDYT?